network/builder/src/builder.rs

```diff
@@ -379,10 +370,26 @@ impl NetworkBuilder {
             None
         };
 
+        // union pubkeys from addrs with pubkeys directly in config
+        let addr_pubkeys_iter = seed_addrs.iter().map(|(peer_id, addrs)| {
+            let pubkey_set: HashSet<_> = addrs
+                .iter()
+                .filter_map(NetworkAddress::find_noise_proto)
+                .collect();
+            (*peer_id, pubkey_set)
+        });
```

Comment:
Should you `filter_map` the outside?  Maybe there's no noise protocol in one.

Reply:
it's correctly handled already. 

if no pubkeys from addr, then pubkey_set is empty HashSet. then, `.extend(pubkey_set)` will just do nothing.

connectivity manager will also just ignore the whole peer entry if no pubkeys from either addrs or config.